quality_check/09-prior_information_matrix.Rmd

```diff
@@ -0,0 +1,225 @@
+---
+title: "Construct _Danio rerio_ prior information matrices"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: J. Taroni for CCDL
+date: 2019
+---
+
+If we want to use `PLIER`, we need to construct prior information matrices
+for zebrafish.
+We'll do that using two sources: 
+[WikiPathways](https://www.wikipathways.org/index.php/WikiPathways) and 
+[ZFIN](https://zfin.org/)
+
+## Setup
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+```{r}
+# BiocManager::install(c("rWikiPathways", "GSA", "org.Dr.eg.db"),
+#                      update = FALSE)
+# install.packages("XML")
+library(rWikiPathways)
+library(XML)
+library(org.Dr.eg.db)
+```
+
+```{r}
+pathway_dir <- file.path("data", "pathways")
+dir.create(pathway_dir, showWarnings = FALSE, recursive = TRUE)
+```
+
+## WikiPathways
+
+```{r}
+# download current version of WikiPathway pathways
+# this is CC0
+downloadPathwayArchive(organism = "Danio rerio", format = "gmt",
+                       destpath = pathway_dir, date = "20190110")
+```
+
+Read the `.gmt` file in and convert to binary matrix
+
+```{r}
+zebrafish_wp_list <- 
+  GSA::GSA.read.gmt(file.path(pathway_dir, 
+                              "wikipathways-20190110-gmt-Danio_rerio.gmt"))
+```
+
+Clean up the list to make it more amenable to the binary prior information
+matrix we'll need for use with PLIER.
+
+```{r}
+names(zebrafish_wp_list$genesets) <- make.names(  # remove hyphens
+  gsub(" ", "_",  # convert spaces to underscores
+       gsub("%WikiPathways_20190110%", " ",  # remove date, WP information 
+            gsub("%Danio rerio", "",  # remove species information
+                 zebrafish_wp_list$geneset.names))
+  )
+)
+# remove pathway names and descriptions
+zebrafish_wp_list <- zebrafish_wp_list$genesets
+```
+
+We're going to use ZFIN identifiers instead of Entrez IDs, which are what are
+used in WikiPathways.
+
+```{r}
+# our goal is to get a binary matrix that represents gene-pathway relationships
+# to do this, we first need to melt the list into a data.frame
+zebrafish_wp_binary_df <- reshape2::melt(zebrafish_wp_list) %>%
+  # get sensible column names
+  dplyr::mutate(Gene = value, Pathway = L1) %>%
+  dplyr::select(Gene:Pathway) %>%
+  # conversion from Entrez IDs to ZFIN IDs with mapIds -- this should be 1:1
+  dplyr::mutate(ZFIN = mapIds(org.Dr.eg.db, 
+                              keys = as.character(Gene), 
+                              keytype = "ENTREZID", column = "ZFIN")) %>%
+  # drop Entrez IDs
+  dplyr::select(-Gene) %>%
+  # drop records missing ZFIN ids
+  dplyr::filter(!is.na(ZFIN)) %>%
+  # unique records only, just in case
```

Comment:
This comment makes me think you don't know for sure if there are unique records or not. Is there a reason you think there might be or could be non-unique records?

Reply:
Point being, if you determine there shouldn't be any reason there'd be duplicates could we just remove this step? But perhaps you are thinking for other species this may be the case?

Reply:
IIRC this is more of an issue when mapping between different gene identifiers then what I settled on here.